gamemodes/terrortown/entities/weapons/weapon_tttbasegrenade.lua

```diff
@@ -290,18 +380,189 @@ function SWEP:Initialize()
+		local grenadey = (is_grenade and target.DamageTriggerBehavior > 0) or (is_grenade_projectile and target.DamageTriggerProjectileExplosion)
+
+		-- has nothing to do with us
+		if not grenadey or target == inflictor
+			-- ignore chain reactions
+			or (target.PreventDamageChainReaction
+			and (hit_by_grenade or hit_by_grenade_projectile)
+			and inflictor:GetShotToActivate())
+		then
+			-- print("ignoring chain reaction [proj] from", inflictor, "on", target)
+			return
+		end
+
+
+		if not target:GetShotToActivate() then
+			target:SetShotToActivate(true)
+			if is_grenade and target.DamageTriggerBehavior == 1 then
+				target:TryCreate(dmginfo:GetAttacker())
+			elseif (is_grenade and target.DamageTriggerBehavior == 2)
+				or (is_grenade_projectile and target.DamageTriggerProjectileExplosion)
+			then
+				target:TryExplode(dmginfo:GetAttacker())
+			end
+		end
+	end)
+end
+
+if CLIENT then
+	---
+	-- @ignore
+	function SWEP:AddToSettingsMenu(parent)
+		local form = vgui.CreateTTT2Form(parent, "header_equipment_additional")
+
+		form:MakeHelp({
+			label = "help_weapon_tttbasegrenade_throw_force",
+		})
+		form:MakeSlider({
+			serverConvar = self:GetGrenadeName() .. "_throw_force",
+			label = "label_weapon_tttbasegrenade_throw_force",
+			min = 0,
+			max = 10,
+			decimal = 2,
+		})
+
+		form:MakeHelp({
+			label = "help_weapon_tttbasegrenade_detonate_timer",
+		})
+		form:MakeSlider({
+			serverConvar = self:GetGrenadeName() .. "_detonate_timer",
+			label = "label_weapon_tttbasegrenade_detonate_timer",
+			min = 0,
+			max = 60,
+			decimal = 2,
+		})
+
+		form:MakeHelp({
+			label = "help_weapon_tttbasegrenade_spherical_physics_radius",
+		})
+		form:MakeSlider({
+			serverConvar = self:GetGrenadeName() .. "_spherical_physics_radius",
+			label = "label_weapon_tttbasegrenade_spherical_physics_radius",
+			min = 0,
+			max = 16,
+			decimal = 0,
+		})
+
+		form:MakeHelp({
+			label = "help_weapon_tttbasegrenade_disarm_hit_players",
+		})
+		form:MakeCheckBox({
+			serverConvar = self:GetGrenadeName() .. "_disarm_hit_players",
+			label = "label_weapon_tttbasegrenade_disarm_hit_players",
+		})
+
+		form:MakeHelp({
+			label = "help_weapon_tttbasegrenade_damage_trigger_behavior",
+		})
+		local dtb = GetConVar(self:GetGrenadeName() .. "_damage_trigger_behavior")
+		dtb = dtb and dtb:GetString() or "0"
+		form:MakeComboBox({
+			label = "label_weapon_tttbasegrenade_damage_trigger_behavior",
+			serverConvar = self:GetGrenadeName() .. "_damage_trigger_behavior",
+			choices = {
+				{
+					title = LANG.TryTranslation("label_weapon_tttbasegrenade_damage_trigger_behavior_none"),
+					value = "0",
+					select = dtb == "0",
+				},
+				{
+					title = LANG.TryTranslation("label_weapon_tttbasegrenade_damage_trigger_behavior_prime"),
+					value = "1",
+					select = dtb == "1",
+				},
+				{
+					title = LANG.TryTranslation("label_weapon_tttbasegrenade_damage_trigger_behavior_instant_explosion"),
+					value = "2",
```

Comment:
remove these translations here. I know why you did that, but this has to be fixed in the UI base.